drivers/xen/xenapi/Config.uk

```diff
@@ -0,0 +1,5 @@
+config LIBXENAPI
+	bool "Xen API headers"
+	select LIBNOLIBC if !HAVE_LIBC
```

Comment:
The headers are unmodified copies from the official Xen sources. I would like to keep it this way so that future updates of the Xen API headers (sync from Xen project) does not need to take modifications into account. Because of this, the dependency is actually correct and needed.

Reply:
Maintainability is definitely a valid point. My only concern is whether such a dependency may make sense in the environment where these headers originate from, but on in our case. Anyway, I see Xen already depends on LIBC in other configs, and most notably in `plat/xen/Config.uk`, so I guess for now it's okay. It would be down to the maintainers of Xen (in Unikraft) to take a decision on whether to depend on a stdlib. So let's leave it as is for now then, until this topic is addressed altogether.

Reply:
I am not fully understand what you mean, because the line says to include `libnolibc` if there is no other `libc` (`!HAVE_LIBC`). I think this is totally reasonable this way to have the required data types for the headers. At least `nolibc` is almost always needed, often because `memcmp` or `memcpy`.

Reply:
Yes exactly, but the problem is that nolibc keeps growing and I feel we haven't defined a clear scope for it, or at least I am not aware of one. For many cases we indlude `nolibc` just for errno-compatible exit codes, and a few primitives like `memcpy()`. Anyway, as said earlier this is out of the scope of this PR, so I think we can resolve this comment and resume discussion in the scope of rearch.

Reply:
`nolibc's` definition is to support pure core repository builds and it should be also dead-code-elimination friendly.